Detectors/ITSMFT/ITS/tracking/src/VertexerTraits.cxx

```diff
@@ -343,22 +347,18 @@ void VertexerTraits::computeTrackletsPureMontecarlo()
 
   for (unsigned int iCurrentLayerClusterIndex{ 0 }; iCurrentLayerClusterIndex < mClusters[0].size(); ++iCurrentLayerClusterIndex) {
     auto& cluster{ mClusters[0][iCurrentLayerClusterIndex] };
-    //const int2 selectedBinsRect{ VertexerTraits::getPhiBins(1, cluster.phiCoordinate, mVrtParams.phiCut) };
     for (unsigned int iNextLayerClusterIndex = 0; iNextLayerClusterIndex < mClusters[1].size(); iNextLayerClusterIndex++) {
       const Cluster& nextCluster{ mClusters[1][iNextLayerClusterIndex] };
-      if (/*GPUCommonMath::Abs(cluster.phiCoordinate - nextCluster.phiCoordinate) < mVrtParams.phiCut &&*/ labelsMC1[iNextLayerClusterIndex] == labelsMC0[iCurrentLayerClusterIndex] && labelsMC1[iNextLayerClusterIndex] != -1
-          /*mEvent->getClusterLabels(0, cluster).getTrackID() == mEvent->getClusterLabels(1, nextCluster).getTrackID() && mEvent->getClusterLabels(0, cluster).getTrackID() != -1*/)
+      if (labelsMC1[iNextLayerClusterIndex] == labelsMC0[iCurrentLayerClusterIndex] && labelsMC1[iNextLayerClusterIndex] != -1)
```

Comment:
Note that the MCCompLabel was recently changed, -1 is not noise!
You should either use 
isValid() (means the label was set and it is not noise)
or isNoise()

Reply:
Ok, I fixed the conversion function to get the labels, unfortunately I am not able to use them directly, due to the GPU's allergy to root-specific keywords. Now they should be consistent again.

Reply:
@mconcas did understand what you mean by root-specific keywords.
Did you finish with commits, shall I merge (with squashing) once the tests are green?

Reply:
Yes, please

Reply:
> @mconcas did understand what you mean by root-specific keywords.
> Did you finish with commits, shall I merge (with squashing) once the tests are green?

It used to be not possible compile with nvcc things that contained root Rtypes and class specific keywords. Now I see that GPURDTypes should fix this. Next iteration I'll streamline this. Not critical, atm.

Reply:
@mconcas : For me the PR is also fine. Just two comments for later:
- For the GPU types, I am including GPUCommonRtypes.h, but as you say this can be fixed later.
- I am right now fixing obtaining the ITS traits from GPUReconstruction, but you can also add this later.

Reply:
Thanks, @davidrohr